### PR TITLE
Added check for empty lines at the beginning and end of a code block

### LINKED ISSUE #81: Add codestyle checks: empty line
There are often empty lines in the code which might be detected automatically:

* after method declaration and before the method-closing-curly-braces:
```
private void doSomething() {

    String myString....

}
```

* after `if`, `for`, `switch`, `case`, `break` statements:
```
if (true) {

    for (;true;) {
    
        switch (value) {

            case 1:

                doSomething();
                break;

        }
    }
}
```

### PR DESCRIPTION
**Pull request by svilenvul**
Closes #81

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    @Test
+    public void verifyMutlitpleEmptyLinesInSwitchWithCases() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(12, MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 29,
+                MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY, 16, MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY, 19,
+                MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 27, MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY);
+        verifyJavaFile("MutlitpleEmptyLinesInSwitchWithCases.java", expectedMessages);
+    }
+
+    private void verifyJavaFile(String testFileName, String[] expectedMessages) throws Exception {
```
Comment (martinvw) [inline src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: Consider also adding a `verifyJavaFileNoErrors`

File: src/test/resources/checks/checkstyle/noEmptyLineSeparatorCheck/EmptyLineInCaseWithBraces.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+public class TestExample {
+    
+    // empty line after class is allowed
+    
+    public void method(){
+        int month = 2;
+        
+        java.util.ArrayList<String> futureMonths =
```
Comment (martinvw) [inline src/test/resources/checks/checkstyle/noEmptyLineSeparatorCheck/EmptyLineInCaseWithBraces.java]: Should I review this code 😉 please add an import in that case

File: src/test/resources/checks/checkstyle/noEmptyLineSeparatorCheck/EmptyLineInCaseWithBraces.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            new java.util.ArrayList<String>();
+        
+        switch (month) {
+            case 1:  {
```
Comment (martinvw) [inline src/test/resources/checks/checkstyle/noEmptyLineSeparatorCheck/EmptyLineInCaseWithBraces.java]: I personally dislike braces in cases 😄

File: src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    @Test
+    public void verifyEmptyLineInSwitchDefinition() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(12, MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 27,
+                MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY);
+        verifyJavaFile("EmptyLineInSwitchDefinition.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInCase() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(13, MSG_LINE_AFTER_OPENING_BRACE_EMPTY);
```
Comment (martinvw) [inline src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: Given there are only a few different message you could even consider to add methods like

```
verifyEmptyLineAfterOpeningBraces(filename, linenumber)
```

**Comment by svilenvul** [comment]
>  martinvw reviewed 2 hours ago 
> PS: should we also make this rule explicit in the coding guidelines for ESH / openHAB?

I am not sure, it seems minor to me. 
@kaikreuzer, WDYT ?

File: src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java (hunk 1 above)
Comment (svilenvul) [inline src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: Thanks, added, I will try to stick to it in the future :)

**Comment by kaikreuzer** [comment]
I agree with @svilenvul, it is nothing important to explicitly mention - people will notice it when running the analysis tool :-)

**Pull request merged**